libs/web/components/src/MarkdownField.tsx

```diff
@@ -6,80 +6,92 @@ import { IntrinsicElements } from "@app-gov/web/types";
-				>
-					Write
-				</span>
-				<span className="mx-2">|</span>
-				<span
-					onClick={onPreviewClick}
-					className={classNames(
-						"hover:text-hero cursor-pointer",
-						preview && "text-hero"
-					)}
-				>
-					Preview
-				</span>
-			</div>
+		const onPreviewClick = useCallback(() => {
+			setPreview(true);
+		}, []);
 
-			<div className="relative flex-1">
-				<textarea
-					{...props}
-					ref={ref}
-					value={value}
-					rows={rows}
-					className="absolute inset-0 resize-none p-2 font-mono text-sm outline-none"
-				/>
+		return (
+			<div
+				className={classNames(
+					className,
+					"border-hero flex min-h-[400px] w-full  flex-col border-2 bg-white"
+				)}
+			>
+				<div className="flex border-b border-b-slate-200 bg-white p-2 text-slate-600">
+					<span
+						onClick={onWriteClick}
+						className={classNames(
+							"hover:text-hero cursor-pointer",
+							!preview && "text-hero"
+						)}
+					>
+						Write
+					</span>
+					<span className="mx-2">|</span>
+					<span
+						onClick={onPreviewClick}
+						className={classNames(
+							"hover:text-hero cursor-pointer",
+							preview && "text-hero"
+						)}
+					>
+						Preview
+					</span>
+				</div>
 
-				<div
-					className={classNames(
-						"absolute inset-0 overflow-scroll bg-white",
-						!preview && "pointer-events-none opacity-0",
-						preview && "!pointer-events-auto !opacity-100"
-					)}
-				>
-					<Markdown className="overflow-auto p-2">
-						{(value as string) ?? ""}
-					</Markdown>
+				<div className="relative flex-1">
+					<textarea
+						{...props}
+						ref={ref}
+						value={value}
+						rows={rows}
+						className="absolute inset-0 resize-none p-2 font-mono text-sm outline-none"
+					/>
+
+					<div
+						className={classNames(
+							"absolute inset-0 overflow-scroll bg-white",
+							!preview && "pointer-events-none opacity-0",
+							preview && "!pointer-events-auto !opacity-100"
+						)}
+					>
+						<Markdown className="overflow-auto p-2">
+							{(value as string) ?? ""}
+						</Markdown>
+					</div>
 				</div>
-			</div>
-			<p className="markdown bg-slate-100 p-2 text-xs">
-				Supported markdown: <strong>**bold**</strong>, <em>*italic*</em>,{" "}
-				<a href="#">[link]()</a>, <code className="!text-xs">`code`</code> and{" "}
-				<span className="mb-4 inline whitespace-pre-wrap bg-slate-50">
-					<code className="!p-0 !text-xs !text-inherit">
-						```multiline code```
-					</code>
+				<span className="flex bg-slate-100 p-2 text-xs">
+					<p className="markdown flex-1">
+						Supported markdown: <strong>**bold**</strong>, <em>*italic*</em>,{" "}
+						<a href="#">[link]()</a>, <code className="!text-xs">`code`</code>{" "}
+						and{" "}
+						<span className="mb-4 inline whitespace-pre-wrap bg-slate-50">
+							<code className="!p-0 !text-xs !text-inherit">
+								```multiline code```
+							</code>
+						</span>
+					</p>
+					<p>{characterCount} / 1024</p>
```

Comment:
should turn red when it exceeds the 1024 limit and we can do character count hook here instead of all the way of side of the form

Reply:
I don't know what you mean by a hook